examples/chip-tool/commands/common/Commands.cpp

```diff
@@ -107,8 +107,14 @@ CHIP_ERROR Commands::RunCommand(int argc, char ** argv)
             return CHIP_ERROR_INVALID_ARGUMENT;
         }
     }
+    argc -= 3;
+    if (!command->InitOptionalArguments(argc, &argv[3]))
+    {
+        ShowCommand(argv[0], argv[1], command);
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
```

Comment:
So here's a question.  Does it make sense to init the required arguments first, then on success ask the command how many non-optional args it ended up with, decrement argc by that, offset by that into argv and call `InitOptionalArguments`?  Then `InitOptionalArguments` won't need an in/out param and the indexing will be much simpler...